### PR TITLE
HC: Lazy leader

### LINKED ISSUE #3929: HC: post commitments on the parent chain - lazy leader
Once there is a block on the parent chain, there should be a corresponding keyblock on the child chain. If there is not in `<X>` time, we consider the leader to be lazy. Then the honest validators are expected to post yet a new commitment on the parent chain with the previous hash (they are replaying their previous commitments).

This is related to #3928 but they do not depend on each other.

### PR DESCRIPTION
**Pull request by velzevur**
Closes #3929

The work on this PR has been supported by ACF.

### CODE REVIEW

File: apps/aecore/src/aec_consensus_hc.erl [hunk 1]
```diff
@@ -99,8 +101,16 @@ start(Config, #{block_production := BlockProduction}) ->
             <<"polling">> :=
                 #{  <<"fetch_interval">> := FetchInterval,
                     <<"nodes">> := Nodes0
-                 } = Polling
-          }} = Config,
+                 } = Polling,
+            <<"producing_commitments">> := ProducingCommitments
+          },
+     <<"time_till_declaring_lazy_leader">> := _TimeTillDeclaringLazy
```
Comment (seanhinde) [inline apps/aecore/src/aec_consensus_hc.erl:107]: Not sure about the name here. "time_until_lazy_leader" maybe, or "lazy_leader_trigger_time"

File: apps/aecore/src/aec_consensus_hc.erl [hunk 2]
```diff
@@ -446,7 +437,8 @@ assert_key_target_range(_) ->
 
 key_header_difficulty(H) ->
     Target = aec_headers:target(H),
-    aeminer_pow:scientific_to_integer(Target).
+    Difficulty = aeminer_pow:scientific_to_integer(Target),
+    Difficulty.
```
Comment (seanhinde) [inline apps/aecore/src/aec_consensus_hc.erl:441]: Variable no longer needed

File: apps/aehttp/src/aehttp_dispatch_ext.erl [hunk 3]
```diff
@@ -67,6 +68,7 @@ run(Queue, F) ->
           end)
     catch
         error:{rejected, _} ->
+            lager:error("AAAAAAAAA Temporary overload", []),
```
Comment (seanhinde) [inline apps/aehttp/src/aehttp_dispatch_ext.erl:71]: AAAAAAARGH perhaps? :)

File: apps/aecore/src/aec_block_insertion.erl [hunk 4]
```diff
@@ -80,7 +80,9 @@ node_prev_key_hash(#node{header = H}) -> aec_headers:prev_key_hash(H).
 node_height(#node{header = H}) -> aec_headers:height(H).
 
 node_difficulty(#node{type = micro}) -> 0;
-node_difficulty(#node{header = H}) -> aec_headers:difficulty(H).
+node_difficulty(#node{header = H}) ->
+    D = aec_headers:difficulty(H),
+    D.
```
Comment (seanhinde) [inline apps/aecore/src/aec_block_insertion.erl:85]: Extra variable

**Review by seanhinde — APPROVED** [review]
Looks good

File: apps/aehttp/src/aehttpc_aeternity.erl [hunk 5]
```diff
@@ -202,7 +214,7 @@ post_request(Path, Body, Host, Port, Timeout) ->
         {ok, {{_, 200 = _Code, _}, _, Res}} ->
             lager:debug("Req: ~p, Res: ~p with URL: ~ts", [Req, Res, Url]),
             {ok, jsx:decode(list_to_binary(Res), [return_maps])};
-        {ok, {{_, 400, _}, _, Res}} ->
+        {ok, {{_ ,400, _}, _, Res}} ->
```
Comment (uwiger) [inline apps/aehttp/src/aehttpc_aeternity.erl:217]: Why?

**Review by uwiger — COMMENTED** [review]
LGTM

**Review by uwiger — APPROVED** [review]

**Pull request merged**